### ISSUE TITLE
Unable to connect

### ISSUE THREAD

**Issue by loibi** [PowerAruba/PowerArubaSW#49]
Got some weird Issues with Connection.

Can connect to the switch.
But when I want to connect again with the same Creds after some Minutes it just says "Unable to connect"
After w8ing for 1 Day it works again for some time
Any Ideas?

**Comment by alagoutte** [comment]
Hi,

What release do you are using ? 
```powershell
Get-Module PowerArubaSW
```
What the switch and firmware ?

There is a "REST Timeout" (300 secs by default if remember), and it is may this problem (but you have a warning on this case...)

You can get this REST Timeout 
```powershell
Get-ArubaSWRestSessionTimeout
```
and configure
```powershell
Set-ArubaSWRestSessionTimeout -timeout 7200
```

**Comment by loibi** [comment]
Hi
0.5
Firmware 16.07
have the default Timeout to 300 Secs

Strange thing - in Switch logs I get "no free REST Sessions available"

Only getting the error when trying to run it from our Monitoring Tool in Background - not in interactive Windows Session.
Also need to be logged out from Web Interface to get it working

**Comment by alagoutte** [comment]
Hi,

Ok ! it is because you don't disconnect your session and it is limited to 5 sessions
Add 
```powershell
Disconnect-ArubaSW (-noconfirm)
```
at the end of your script

or look to reuse the same session (look $DefaultArubaSWConnection global variable)

**Comment by loibi** [comment]
Hi.
Tried that already, but still got same behaviour.
Looks like some session Problems?
________________________________

**Comment by alagoutte** [comment]
I try and get the same issue when launch more of 5 sessions...
but if i use _Disconnect-ArubaSW_ it is working for me (with 16.06 firmware)

Do you are sure don't have forget to disconnect on your code ?

**Comment by loibi** [comment]
Tried – still getting the same ☹

Using the Script from interactive Windows RDP Session -> OK
Using it on the same Server via the Monitoring Software -> Unable to Login and no rest Sessions in Events

Dont understand why
________________________________
Von: Alexis La Goutte <notifications@github.com>
Gesendet: Tuesday, November 13, 2018 3:24:38 PM
An: PowerAruba/PowerArubaSW
Cc: loibi; Author
Betreff: Re: [PowerAruba/PowerArubaSW] Unable to connect (#49)


I try and get the same issue when launch more of 5 sessions...
but if i use Disconnect-ArubaSW it is working for me (with 16.06 firmware)

Do you are sure don't have forget to disconnect on your code ?

**Comment by loibi** [comment]
I think I found something.

When it does not work:
Getting 2 Event entries with xx logged in from REST on the switch.
But Output from the Script is:
VERBOSE: POST http://switch/rest/v4/login-sessions with -1-byte payload
VERBOSE: received -1-byte response of content type application/json
Unable to connect

So Login ist ok but then something happens and whole script breaks?

When this happens 3 Times then I am above the 5 Sessions because script does not come to the disconnect at the end

________________________________
Von: Andreas Loibingdorfer <loibi2@hotmail.com>
Gesendet: Tuesday, November 13, 2018 5:23:17 PM
An: PowerAruba/PowerArubaSW
Betreff: AW: [PowerAruba/PowerArubaSW] Unable to connect (#49)

Tried – still getting the same ☹

Using the Script from interactive Windows RDP Session -> OK
Using it on the same Server via the Monitoring Software -> Unable to Login and no rest Sessions in Events

Dont understand why
________________________________
Von: Alexis La Goutte <notifications@github.com>
Gesendet: Tuesday, November 13, 2018 3:24:38 PM
An: PowerAruba/PowerArubaSW
Cc: loibi; Author
Betreff: Re: [PowerAruba/PowerArubaSW] Unable to connect (#49)


I try and get the same issue when launch more of 5 sessions...
but if i use Disconnect-ArubaSW it is working for me (with 16.06 firmware)

Do you are sure don't have forget to disconnect on your code ?

**Comment by alagoutte** [comment]
Ok strange...

What it your Monitoring Software ?
Do you have change some stuff because we use v3 for login-sessions actually on PowerArubaSW code

**Comment by loibi** [comment]
tried to change it to v4 for testing, no difference.
you doing 2 logins at connect? cause i get 2 in the event log
________________________________

**Comment by alagoutte** [comment]
No, there is no 2 connect

there is some other query (get) but can be disable with -noverbose

With next version (0.6), there is a enhance to be more fast (via Disable HTTP KeepAlive...)

**Comment by alagoutte** [comment]
Can you send me a email (my email is available on the top of ps1 file like https://github.com/PowerAruba/PowerArubaSW/blob/master/PowerArubaSW/Public/Connection.ps1 )

**Comment by alagoutte** [comment]
Patch for add -usebasicparsing and enhance info when fail to connect are merged and it will be available on 0.6
